### ISSUE TITLE
Nested messages not locking

### ISSUE THREAD

**Issue by twoism** [nilslice/protolock#47]
It seems that nested messages are not currently support unless I have missed something.

**Repro**
```protobuf
syntax = "proto3";

package test;

message Outer {
    message Inner {
        string name = 1;
    }

    Inner inner_msg = 1;
}
```

```bash
protolock init && protolock commit && protolock status
```

**Change the the `name` field identifier.**

```protobuf
message Outer {
    message Inner {
        string name = 2;
    }

    Inner inner_msg = 1;
}
```
Running `protolock status` will exit cleanly. Is this a known issue or is there planned support for nested messages? I can take a stab at adding it if not.

Thanks!

**Comment by nilslice** [comment]
Hey @twoism - thank you for reporting this! I was sure nested messages were accurately accounted for, but without being able to run through your example at the moment, I'll have to get back to you later.

If you wanted to see where some of this was implemented, check out the tests around here: https://github.com/nilslice/protolock/blob/master/rules_test.go#L803-L820

If you end up finding the root of the issue and have interest and time to PR something, that would be fantastic. We have tried to keep a strong test suite against all the rules, so if any code added could impact or require additional testing, we would appreciate having those tests included in the PR, or to have work with us together to get them written.

cc/ @celrenheit, who kindly contributed lots of the nested message checking.

**Comment by nilslice** [comment]
@twoism - no pressure, just wanted to know if you anticipated having time or interest in digging in here. If not, I will take a closer look this week.

Thanks!

**Comment by nilslice** [comment]
@twoism - to keep you apprised on this, I have a branch with a partial fix. There are a few rule funcs that do properly check for nested message changes, but a couple that do not. So, I'll push the branch and ping you once it's up for review and if you'd like to test against your proto set that would be awesome. Thanks again for pointing this out!

**Comment by petervandenbroek** [comment]
@nilslice Do you have any update on this fix?
We use nested messages a lot and it would really help to ensure no breaking changes can be made to those.

Thanks in advance :)

**Comment by prernagoyal** [comment]
Hey @nilslice - Do we have an update on this? 

Thanks!

**Comment by nilslice** [comment]
@prernagoyal - nothing yet.. unfortunately it will take me some time before I can dig into this further. If you would like to help, modifying the existing tests to add cases where this fails when it is expected to succeed would be awesome.

**Comment by nilslice** [comment]
@twoism - if you'd like to test, #122 has a fix for the nested message issue. Let me know if you'd like to run through your protos as a test case prior to merge!

Your initial example is now covered: 
`CONFLICT: "Outer.Inner" field: "name" has a different ID: 2, previously 1 [test.proto]`

**Comment by prernagoyal** [comment]
> If you would like to help, modifying the existing tests to add cases where this fails when it is expected to succeed would be awesome.
@nilslice  Would it help if I raise another PR based off of master which only modifies the existing tests to have nested messages and that those tests pass with the older version of the code even with changing the nested messages? We won't merge that PR though.

**Comment by nilslice** [comment]
This has been fixed and released in [v0.15.0](https://github.com/nilslice/protolock/releases/tag/v0.15.0)!
